docs/day-2-operations/Backups.md

```diff
@@ -0,0 +1,272 @@
+
+## Data Sets
+This section goes into more detail on the different data sets in the different persistent stores, their purposes, the storage type used and if it is the "source of truth" or not.
+
+The full state of a 3scale deployment is stored across these services and their persistent volumes:
+
+* `system-mysql`:  Mysql database (`mysql-storage`)
+* `system-storage`: Volume for Files
+* `zync-database`: Postgres database for `zync` component. This uses "HostPath" as storage. If the pod is moved into another node the data is lost. That is not a problem because the data are sync jobs and does not need to be 100% persistent.
+* `backend-redis`: Redis database (`backend-redis-storage`)
+* `system-redis`: Redis database (`system-redis-storage`)
+
+### `system-mysql`
+This is a relational database for storing the information about users, accounts, APIs, plans etc in the 3scale Admin Console. 
+
+A subset of this information related to Services is synchronized to the `Backend` component and stored in `backend-redis`. `system-mysql` is the source of truth for this information. 
+
+### `system-storage`
+This is for storing files to be read and written by the `System` component. They fall into two categories:
+
+* Configuration files read by the System component at run-time
+* Static files (HTML, CSS, JS, etc) uploaded to System by it's CMS feature, for the purpose of creating a Developer Portal.
+
+Note that `System` can be scaled horizontally with multiple pods uploading and reading said static files, hence the need for a `RWX` PersistentVolume.
+
+### `zync-database`
+This is a relational database for storing information related to the synchronization of identities between 3scale and an Identity provider.
+This information is not duplicated in other components and this is the sole source of truth.
+
+### `backend-redis`
+This contains multiple data sets used by the `Backend` component:
+
+* Usages: This is API usage information aggregated by `Backend`. It is used by `Backend` for rate-limiting decisions and by `System` to display analytics information in the UI or via API.
+* Config: This is configuration information about Services, Rate-limits, etc that is synchronized from `System` via an internal API. This is NOT the source of truth of this info, `System` and `system-mysql` is.
+* AuthKeys: Storage of OAuth keys created directly in `Backend`. This is the source of truth for this information.
+* Queues: Queues of Background Jobs to be executed by worker processses. These are ephemeral and are deleted once processed.
+
+### `system-redis`
+This contains Queues for jobs to be processed in background. These are ephemeral and are deleted once processed.
```

Comment:
Some jobs are not duplicated in other components. Making this database its the sole source of truth.

Reply:
So backup is needed right? The restore could be tricky and lead to inconsistency @hallelujah ?

Reply:
Yes this needs backup. Inconsistencies could happen like sending email twice on restore.
I just verified, billing jobs do not lead to inconsistencies.

Reply:
"sole source of truth" .... for pending jobs, right?
Once executed, they are removed I understand.

So, we go with the backup of it, and risk that some tasks could be done twice?
If there is an enhancement needed here to have System avoid that (maybe for some jobs more important than others) then let's create the JIRA issue for that for the future.

Reply:
Yes when a job is executed then it is removed. It would be extremely difficult to control that kind of Job executing only once due to the parallelism execution of those jobs.

Reply:
@hallelujah so for my understanding there is no need for a jira issue to treat this, as it will be "extremely difficult" right?